LISTENER_MIGRATION.md

```diff
@@ -0,0 +1,338 @@
+# Migration Guide
```

Comment:
random: I wonder if we should merge this into the v4 migration guide instead of keeping them separate?

Reply:
Agreed. Will update.

Reply:
Realizing now that might be a little confusing - having multiple code snippets for each flow that aren't compatible. Do you think we should replace the current migration guide code snippets/documentation with this new implementation pattern? Otherwise we could rename this guide to v4.9.0_MIGRATION_GUIDE and include more description about how/why to migrate to this pattern?

Reply:
I think what I'll do is move all of the still relevant portions of the migration guide, and the new listener pattern snippets to a `v4.9_MIGRATION_GUIDE` and then link to that new guide from the README and the old migration guide (I don't want to lost the old snippets for merchants stuck in the middle)

Reply:
That's a really good idea actually `v4.9.0_MIGRATION_GUIDE` I like that it's explicit about the affected version. We'll have an opportunity to clean up the migration guides with the next major version too.